src/executor/evaluator.rs

```diff
@@ -45,7 +45,7 @@ impl BoundExpr {
             }
             BoundExpr::IsNull(expr) => {
                 let array = expr.expr.eval(chunk)?;
-                Ok(ArrayImpl::Bool(
+                Ok(ArrayImpl::new_bool(
```

Comment:
ArrayImpl::Bool(<the original BoolArray>.into())

Reply:
Leaving this as it uses `collect` to construct array.